.azure-pipelines/util/download-build-steps.yml

```diff
@@ -0,0 +1,9 @@
+parameters:
+  osName: ''
+
+steps:
```

Comment:
Looks like a number of these templates can be folded into the one yaml file to avoid some levels of complexity.

Reply:
Since these are reused between different jobs in different pipelines, it makes sense to have a separate template. The primary issue is that you cannot template a job matrix. Meaning, I can't make a job template where a strategy is conditionally defined.

Reply:
Yes that is one of the limitations of the templates. However I don't think that would be necessary if you had one matrix strategy for both and just control the framework as one of the matrix variables.